### PR TITLE
Disable Tor if incognito mode is disabled by group policy.

### PR DESCRIPTION
**Pull request by boocmp**
Resolves https://github.com/brave/brave-browser/issues/31434

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. In admin cmd execute 
    WIN ```REG ADD HKLM\SOFTWARE\Policies\BraveSoftware\Brave /v IncognitoModeAvailability /t REG_DWORD /d 1```
    MAC ```defaults write com.brave.Browser IncognitoModeAvailability -integer 1```
2. Launch brave, check there is no 'New private window with Tor' in main menu and Tor settings on brave://settings/privacy are disabled (except Snowflake extension)
3. Return default policy value
   WIN ```REG ADD HKLM\SOFTWARE\Policies\BraveSoftware\Brave /v IncognitoModeAvailability /t REG_DWORD /d 0```
   MAC ```defaults write com.brave.Browser IncognitoModeAvailability -integer 0```
4. Relaunch brave, check tor settings are enabled.

### CODE REVIEW

File: browser/policy/brave_policy_browsertest.cc [hunk 1]
```diff
@@ -81,7 +81,7 @@ class TorEnabledPolicyBrowserTest : public BravePolicyTest {
 IN_PROC_BROWSER_TEST_F(TorEnabledPolicyBrowserTest, TorDisabledPrefValueTest) {
   // When policy is set, explicit setting doesn't change its pref value.
   TorProfileServiceFactory::SetTorDisabled(true);
-  EXPECT_FALSE(TorProfileServiceFactory::IsTorDisabled());
+  EXPECT_FALSE(TorProfileServiceFactory::IsTorDisabled(browser()->profile()));
 }
```
Comment (goodov) [inline browser/policy/brave_policy_browsertest.cc:85]: a test with incognito disabled/forced?

File: browser/resources/settings/brave_routes.ts [hunk 2]
```diff
@@ -31,7 +31,11 @@ export default function addBraveRoutes(r: Partial<SettingsRoutes>) {
     r.REWARDS = r.BASIC.createSection('/rewards', 'rewards')
   }
   if (loadTimeData.getBoolean('areShortcutsSupported')) {
-    r.SHORTCUTS = r.SYSTEM.createChild('/system/shortcuts')
+    if (r.SYSTEM) {
+      r.SHORTCUTS = r.SYSTEM.createChild('/system/shortcuts')
+    } else if (!isGuest) {
+      console.error('[Brave Settings Overrides] Routes: could not find SYSTEM page')
+    }
```
Comment (goodov) [inline browser/resources/settings/brave_routes.ts:38]: what this change is for?

File: browser/resources/settings/brave_tor_page/brave_tor_subpage.html [hunk 3]
```diff
@@ -37,7 +37,7 @@
   sub-label="$i18n{torEnabledDesc}" disabled="[[disableTorOption_]]"
   on-settings-boolean-control-change="onTorEnabledChange_">
 </settings-toggle-button>
-<settings-toggle-button pref="{{prefs.tor.auto_onion_location}}" class="cr-row" label="$i18n{autoOnionLocationLabel}"
+<settings-toggle-button id="autoOnionLocation" pref="{{prefs.tor.auto_onion_location}}" class="cr-row" label="$i18n{autoOnionLocationLabel}"
```
Comment (goodov) [inline browser/resources/settings/brave_tor_page/brave_tor_subpage.html:40]: what this change is for?

File: browser/ui/webui/settings/brave_tor_handler.cc [hunk 4]
```diff
@@ -390,22 +390,27 @@ void BraveTorHandler::IsTorEnabled(const base::Value::List& args) {
...
-                      base::Value(!TorProfileServiceFactory::IsTorDisabled()));
+                      base::Value(!TorProfileServiceFactory::IsTorDisabled(
+                          web_ui()->GetWebContents()->GetBrowserContext())));
   }
 }
 
 void BraveTorHandler::IsTorManaged(const base::Value::List& args) {
   CHECK_EQ(args.size(), 1U);
 
+  const bool is_available = TorProfileServiceFactory::IsTorAvailable(
```
Comment (goodov) [inline browser/ui/webui/settings/brave_tor_handler.cc:408]: let's remove `IsTorAvailable` and add `IsTorManaged` instead.

right now `IsTorAvailable` and `IsTorDisabled` contradict each other.

File: browser/resources/settings/brave_tor_page/brave_tor_subpage.html (hunk 3 above)
Comment (boocmp) [reply browser/resources/settings/brave_tor_page/brave_tor_subpage.html:40]: for tests

File: browser/resources/settings/brave_routes.ts (hunk 2 above)
Comment (boocmp) [reply browser/resources/settings/brave_routes.ts:38]: Just found a bug in the Guest Window.

File: browser/policy/brave_policy_browsertest.cc (hunk 1 above)
Comment (boocmp) [reply browser/policy/brave_policy_browsertest.cc:85]: BraveTorTestWithCustomProfile.Incognito checks all states now.

File: browser/profiles/profile_util.cc [hunk 5]
```diff
@@ -46,7 +46,8 @@ bool IsGuestProfile(content::BrowserContext* context) {
 
 bool IsTorDisabledForProfile(Profile* profile) {
 #if BUILDFLAG(ENABLE_TOR)
-  return TorProfileServiceFactory::IsTorDisabled() || profile->IsGuestSession();
+  return TorProfileServiceFactory::IsTorDisabled(profile) ||
+         profile->IsGuestSession();
```
Comment (goodov) [inline browser/profiles/profile_util.cc:50]: I wonder if `profile->IsGuestSession();` should now be moved to `TorProfileServiceFactory::IsTorDisabled`?

Comment (boocmp) [reply browser/profiles/profile_util.cc:50]: Looks like it can

**Review by goodov — APPROVED** [review]

**Review by fmarier — APPROVED** [review]
I did not look through the actual code changes, but disabling Tor windows when incognito is OFF is the right thing to do.

**Comment by LaurenWags** [comment]
Verification in progress with

```

```

**Pull request merged**